server/api.go

```diff
@@ -554,6 +554,20 @@ func LogMiddleware(typ string, opts LogOptions) func(next http.Handler) http.Han
 	}
 }
 
+// RecoverMiddware is middleware that logs and returns a 500 error if something unhandled (panic) occurs
+func RecoverMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			err := recover()
+			if err != nil {
```

Comment:
A minor thing, recover does not necessarily return an error, so therefore you often see `r := recover()`.